ui/ui-components/pages/model/[modelId]/source/[sourceId]/edit.tsx

```diff
@@ -537,6 +551,7 @@ const Page: NextPage<Props> = ({
                               placeholder={opt.placeholder}
                               name={`source.options.${opt.key}`}
                               {...register(`source.options.${opt.key}`, {
+                                shouldUnregister: true,
```

Comment:
On the first few renders until we get back the connectionOptions response, it falls back to a disabled text input. This was causing problems because register by default does not unregister on unmount, so sometimes the custom `onChange` function passed here stuck around and was still getting called whenever we changed the value on the typeahead. Passing `shouldUnregister: true` here correctly cleans up the old onChange on unmount as expected.